rules_poetry/poetry.bzl

```
@@ -124,7 +169,23 @@ def _impl(repository_ctx):
         if requested.lower() in excludes:
             fail("pyproject.toml dependency {} is also in the excludes list".format(requested))
 
+    toml_markers = {}
+
+    if metadata["lock-version"] in ["2.0"]:
+        poetry_deps = mapping['pyproject']['tool']['poetry']
+        
+        for dep in poetry_deps['dependencies']:
+            extract_markers(repository_ctx, toml_markers, dep, poetry_deps['dependencies'][dep])
```

```suggestion
        for dep, markers in poetry_deps['dependencies'].items():
            extract_markers(repository_ctx, toml_markers, dep, markers)
```

nit: similar to @anthonyjpratti's comments, slightly cleaner this way IMO

---

added better handling and use of items() to make this cleaner [7aa2a09](https://github.com/mongodb-forks/rules_poetry/pull/1/commits/7aa2a09847cc9482b937f4381021a9cb45946821)